### PR TITLE
New Project UI

### LINKED ISSUE #109: As a registered user, I want to create a project so that I can begin my work

### LINKED ISSUE #144: Create the Workspaces view (see mockup)

### PR DESCRIPTION
**Pull request by nicksalt**
Adds UI to new project creation flow based on Figma mockups

Closes #144 
Closes #109

### CODE REVIEW

File: Datahub/src/Datahub.Core/Model/Datahub/DatahubProjectDBContext.cs [hunk 1]
```diff
@@ -178,5 +178,9 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         modelBuilder.Entity<Project_Current_Monthly_Cost>()
             .Property(mc => mc.Id)
             .ValueGeneratedOnAdd();
+
+        modelBuilder.Entity<Datahub_Project_User>()
+            .Property(u => u.User_ID)
+            .ValueGeneratedOnAdd();
```
Comment (yjmrobert) [inline Datahub/src/Datahub.Core/Model/Datahub/DatahubProjectDBContext.cs]: We don't want to generate this value, this is the user's Azure GUID

File: Datahub/src/Datahub.Portal/Components/Projects/UserProjects.razor [hunk 2]
```diff
@@ -92,5 +111,16 @@
 
         return userRole.IsDataApprover ? "Data Approver" : "Member";
     }
+    
+    private async Task ShowProjectDialog()
+    {
+        var options = new DialogOptions { CloseOnEscapeKey = true };
+        var dialog = await DialogService.ShowAsync<CreateProjectDialog>("Delete Server", options);
```
Comment (yjmrobert) [inline Datahub/src/Datahub.Portal/Components/Projects/UserProjects.razor]: I think we should rename this dialog and put localization on the title 😄

File: Datahub/src/Datahub.Portal/Views/Modals/CreateProjectDialog.razor [hunk 3]
```diff
@@ -54,15 +64,20 @@
     private MudTextField<string>? _acronymField;
 
     private bool _success;
-    
+    private bool _isCreating;
+
+    protected override void OnInitialized()
+    {
+        MudDialog.SetTitle(Localizer[Title]);
```
Comment (yjmrobert) [inline Datahub/src/Datahub.Portal/Views/Modals/CreateProjectDialog.razor]: Ah, here's where it's setting. Are we able to set this when we show dialog instead?

File: Datahub/src/Datahub.Core/Model/Datahub/DatahubProjectDBContext.cs (hunk 1 above)
Comment (nicksalt) [reply Datahub/src/Datahub.Core/Model/Datahub/DatahubProjectDBContext.cs]: Hey @yjmrobert , sorry meant to make the value generated the ID for this table... Thanks for catching this

File: Datahub/src/Datahub.Portal/Views/Modals/CreateProjectDialog.razor (hunk 3 above)
Comment (nicksalt) [reply Datahub/src/Datahub.Portal/Views/Modals/CreateProjectDialog.razor]: In the updated commits, I changed where the dialog title is set,

**Pull request merged**